### PR TITLE
Pol 102 google instance stack driver utilization policy

### PR DESCRIPTION
**Pull request by nshivappa**
### Description

This Policy Template gathers Google StackDriver utilization for instances on 30-day intervals and resizes them after user approval. This is meant to be run as a monthly policy.

### Issues Resolved

- Removed the tagging escalation/action
- Added parameters for Avg CPU and memory
- Added Resize action for underutilized instances
- Changed the authentication to credential services
- changed name of the policy to Google Inefficient Instance Utilization using StackDriver

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### Running template link

https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e15d8a47abe69000112a401

### Screenshot

image

### CODE REVIEW

File: cost/google/instances_stackdriver_utilization/README.md [hunk 1]
```diff
@@ -1,36 +1,51 @@
-## Google Instance StackDriver Utilization Policy
+# Google Inefficient Instance Utilization using StackDriver
 
-### What it does
+## What it does
 
-This Policy Template gathers Google StackDriver utilization for instances on 30 day intervals. This is meant to be run as a monthly policy. 
+This Policy Template gathers Google StackDriver utilization for instances on 30 day intervals and resizes them after user approval. This is meant to be run as a monthly policy.
```
Comment (cdwilhelm) [inline cost/google/instances_stackdriver_utilization/README.md]: please remove This is meant to be run as a monthly policy.

File: cost/google/instances_stackdriver_utilization/google_instances_stackdriver_utilization.pt [hunk 2]
```diff
@@ -151,10 +157,126 @@ datasource "ds_disk_utilization" do
...
+        var current_instance=res[res.length-1].trim();
+        if(current_instance != null && !_.contains(current_instance, "custom")){
+          var next_instance_size = ds_google_instance_size_map[current_instance]["down"] ;
+          var next_machine_type= instance_data["machine_type"].replace(current_instance, next_instance_size);
+        }
+        if(next_instance_size == null && typeof next_instance_size == "undefined" ){
+          next_instance_size="N/A";
+          next_machine_type="N/A";
+        }
+        results.push({
```
Comment (cdwilhelm) [inline cost/google/instances_stackdriver_utilization/google_instances_stackdriver_utilization.pt]: please sort by region.

File: cost/google/instances_stackdriver_utilization/README.md (hunk 1 above)
Comment (nshivappa) [reply cost/google/instances_stackdriver_utilization/README.md]: Updated

File: cost/google/instances_stackdriver_utilization/google_instances_stackdriver_utilization.pt (hunk 2 above)
Comment (nshivappa) [reply cost/google/instances_stackdriver_utilization/google_instances_stackdriver_utilization.pt]: Added _.sortBy function for result.

**Comment by cdwilhelm** [comment]
This looks good.  I will approve next week.

**Pull request merged**